panel/src/components/Views/ModelView.vue

```diff
@@ -57,16 +62,34 @@ export default {
 		}
 	},
 	mounted() {
+		this.onInput = debounce(this.onInput, 50);
```

Comment:
`Thought:` Instead of debouncing here the `onInput` method, I think we should only debounce the API request. That way we could avoid issues we currently have (e.g. the not-so-responsive range slider due to debouncing updating the values) and still debounce the API calls. Though, this might be hard to realize when the content comes only from the backend through the props (might be further delayed) and not directly updated via un-debounced input event.

This might be actually a **bigger concern**: How much delay do we add by having the updated values coming in via props from the backend? Most of our inputs don't work with a local state but rely on that the new value they emit via `input` event, immediately loops in via the `value` prop again.

Reply:
Debouncing the API request is happening on the next branch in one of the next steps and it works really well. 

Working on this, I realized that we only need the debouncing here to avoid multiple JSON.stringify comparisons between original and changed fields. This has been the big performance hit. It really is all about showing or hiding the form buttons. 

So I think it might be worth it to separate the "hasChanges" check into its own method and really only debounce that one as well. 

Inputs could then fire right away. Autosaving via API requests would be debounced and the changes checks would also be debounced. By separating those into two debounced methods, we can have more control over ideal debounce intervals to get the feeling right. It seems like it's totally enough to send API requests every 200 ms or something. But we really might want to have the update checks more frequently.

Reply:
Sounds good. 

How do we avoid the delay for the updated values coming in via the props though?

Reply:
I think we really can't. This is a problem that we haven't solved at all so far and will be a new "feature" anyway. In the next branch, I'm instantly overwriting the view props on input and before submit. When the API requests finish and the backend returns different values, we could probably compare the differences first and do nothing if there are no changes. If there are changes, we might actually think about triggering a dialog or a little notification or something, but we really need to try this out.